cpp/include/raft/spatial/knn/detail/ann_kmeans_balanced.cuh

```diff
@@ -522,45 +574,47 @@ void balancing_em_iters(const handle_t& handle,
 }
 
 /** Randomly initialize cluster centers and then call `balancing_em_iters`. */
-template <typename T>
+template <typename T, typename IdxT, typename LabelT>
 void build_clusters(const handle_t& handle,
```

Comment:
Please provide the compatible overload/wrapper for `predict` and `build_clusters` with the previous api (with no `dataset_norm` argument). I have a number of usages of these two in #789 scattered across the build module.

Reply:
So, in those cases `predict` is only invoked once with the same dataset, hence it doesn't make any difference to compute the norm in `predict` as opposed to pre-computing?

Reply:
Yep. Depending on the mode of operation, there is a loop over either clusters or subspaces of the original/transformed dataset.

Reply:
+1 for Artem's request, since it simplifies the code on the call site.